test/k8sT/assertionHelpers.go

```diff
@@ -32,6 +32,9 @@ func ExpectKubeDNSReady(vm *helpers.Kubectl) {
 func ExpectCiliumReady(vm *helpers.Kubectl) {
 	err := vm.WaitforPods(helpers.KubeSystemNamespace, "-l k8s-app=cilium", 600)
 	ExpectWithOffset(1, err).Should(BeNil(), "cilium was not able to get into ready state")
+
+	err = vm.CiliumPreFlightCheck()
```

Comment:
Several conditions inside of this function are not guaranteed to be met right after the pod has become ready. We need retries until a timeout has been reached. You can consider using the new `WaitUntil`:

```
import "github.com/cilium/cilium/pkg/testutils"
[...]
successful := testutils.WaitUntil(func() bool { return condition() }, time.Minute)
```

You also mention that you want to make this async with a WaitGroup. I'm not sure I understand that aspect.

On failure, this should also print `cilium status` and `cilium-health status` of all agents

Reply:
Even with `--probe`, it is not guaranteed that we have already discovered all nodes and all other nodes are already running the health server. Retries will not make anything slower at all, if it is already in place, it will return right away. We can only test conditions without a timeout that are guaranteed to be in place and the cluster health is completely disconnected from the ready state of the cilium pod. The cilium pod is ready as soon as the pod is functional from a local node scope perspective.

Reply:
Did the change in the helper, should be ready for review.